tests/components/essent/test_sensor.py

```diff
@@ -0,0 +1,445 @@
+"""Test the Essent sensors."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+from unittest.mock import AsyncMock
+
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+
+from essent_dynamic_pricing.models import EnergyData, EssentPrices
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util import dt as dt_util
+
+from homeassistant.components.essent.sensor import (
+    _format_dt_str,
+    _parse_tariff_datetime,
+)
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+from . import setup_integration
+
+pytestmark = [
+    pytest.mark.usefixtures("entity_registry_enabled_by_default"),
+    pytest.mark.freeze_time("2025-11-16 10:30:00+01:00"),
+]
+
+
+async def _enable_unique_ids(
+    hass: HomeAssistant,
+    entry: MockConfigEntry,
+    ent_reg: er.EntityRegistry,
+    unique_ids: tuple[str, ...],
+) -> None:
+    """Enable disabled entities and reload when needed."""
+    updated = False
+    for unique_id in unique_ids:
+        entity_id = ent_reg.async_get_entity_id("sensor", "essent", unique_id)
+        assert entity_id is not None
+        reg_entry = ent_reg.async_get(entity_id)
+        assert reg_entry is not None
+        if reg_entry.disabled_by:
+            ent_reg.async_update_entity(entity_id, disabled_by=None)
+            updated = True
+
+    if updated:
+        await hass.config_entries.async_reload(entry.entry_id)
+        await hass.async_block_till_done()
```

Comment:
For enabling entities, you can just use `@pytest.mark.usefixtures("entity_registry_enabled_by_default")`